app/models/package_manager/base/api_version_to_upsert.rb

```diff
@@ -22,14 +22,17 @@ def initialize(
       end
 
       def to_version_model_attributes
-        {
+        non_nillable_attributes = {
           number: @version_number,
           published_at: @published_at,
           runtime_dependencies_count: @runtime_dependencies_count,
           original_license: @original_license,
           repository_sources: @repository_sources,
-          status: @status,
         }.compact
+
+        # If the status is currently "Removed" and a change comes in to update that status to nil/"Active",
```

Comment:
I'm not sure if this is something that needs to take the PackageManager into consideration, since it could potentially be unintentionally setting a bunch of statuses to nil if the other PackageManagers are relying on this cleanup to avoid updating stuff that don't need to be updated, vs passing in exactly what needs to be updated. But at least from `Base#update`, if the status gets set to nil here, it'll get set back to 'Removed' right after this in the `deprecate_version` call.

This adds another reason to change the idea of "version is active" from status: nil to status: 'active' -- we can't tell if the nil is because we haven't looked at the version status so it's 'active' by default, or if it's because it's actually active based off of some other piece of information we get about the version, or if it's a bug somewhere upstream. But that will be a bigger change that touches all the package managers, and out of scope for this PR.

Reply:
Agreed on the nil status problem. I think this change is reasonably safe because the pre-refactor version of `add_version` also always overwrote the status field

```
      version_hash = version_hash.symbolize_keys.slice(
        :number,
        :published_at,
        :runtime_dependencies_count,
        :original_license,
        :repository_sources,
        :status
      )

      existing = db_project.versions.find_or_initialize_by(number: version_hash[:number])
      existing.skip_save_project = true
      existing.assign_attributes(version_hash)
```